File: testdata/bench/javascript/BenchmarkTest00017.js
// Vulnerable: SQL injection via req.params in template literal
const express = require('express');
const db = require('./db');
const router = express.Router();
router.get('/users/:role', (req, res) => {
    const role = req.params.role;
    const sql = `SELECT name, email FROM users WHERE role = '${role}'`;
    db.query(sql, (err, results) => { res.json(results); });
});
module.exports = router;

File: testdata/bench/rust/BenchmarkTest00017.rs
use actix_web::{web, HttpResponse};
use std::collections::HashMap;

// Vulnerable: SQL injection via to_string propagation
async fn handler(query: web::Query<HashMap<String, String>>) -> HttpResponse {
    let param = query.get("term").unwrap_or(&String::new()).clone();
    let s = param.to_string();
    let sql = format!("SELECT * FROM products WHERE description LIKE '%{}%'", s);
    // sqlx::query(&sql).execute(&pool).await
    HttpResponse::Ok().body(sql)
}
